weave/compile.py

```diff
@@ -87,6 +87,39 @@ def _quote_node(node: graph.Node) -> graph.Node:
     return weave_internal.const(node)
 
 
+def _node_simplification(node: graph.Node) -> typing.Optional[graph.OutputNode]:
+    if isinstance(node, graph.OutputNode) and node.from_op.name == "count":
```

Comment:
Needs comments.

Reply:
oh yeah, i'll make this much more beautiful